### PR TITLE
Add vc support

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Describe your changes

Add vc support.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

**Review by devang-ml — COMMENTED** [review]
- Please update options.md

**Comment by guotuofeng** [comment]
could the description be more details? I don't know the background so there is no way to review it.

File: olive/systems/system_config.py [hunk 1]
```diff
@@ -15,6 +15,16 @@
 from olive.systems.common import AzureMLDockerConfig, LocalDockerConfig, SystemType
 
 
+class VCConfig(ConfigBase):
+    instance_type: str
+    interactive: bool = False
+    imageVersion: str = ""
+    slaTier: str = "Standard"
+    priority: str = "Medium"
+    tensorboardLogDirectory: str = "/scratch/tensorboard_logs"
```
Comment (guotuofeng) [inline olive/systems/system_config.py:24]: default to None or required instead of the hardcode value?

File: olive/systems/system_config.py [hunk 2]
```diff
@@ -15,6 +15,16 @@
 from olive.systems.common import AzureMLDockerConfig, LocalDockerConfig, SystemType
 
 
+class VCConfig(ConfigBase):
```
Comment (guotuofeng) [inline olive/systems/system_config.py:18]: do we need add documents/comments about the meaning of these fields?

File: olive/systems/system_config.py (hunk 1 above)
Comment (xiaoyu-work) [reply olive/systems/system_config.py:24]: I'm testing the None value for these attributes now. If it supports None and the API can map a default value, I will update this logic.

File: olive/systems/system_config.py (hunk 2 above)
Comment (xiaoyu-work) [reply olive/systems/system_config.py:18]: I will send the doc link to you offline.

File: olive/systems/system_config.py (hunk 1 above)
Comment (guotuofeng) [inline olive/systems/system_config.py:24]: make the naming consistent by all use snake case(variable_part) instead of came case?

Comment (xiaoyu-work) [reply olive/systems/system_config.py]: I updated the PR and removed all other configs.

File: olive/systems/azureml/aml_system.py [hunk 3]
```diff
@@ -81,7 +83,10 @@ def __init__(
         self._assert_not_none(aml_docker_config)
         aml_docker_config = validate_config(aml_docker_config, AzureMLDockerConfig)
         azureml_client_config = validate_config(azureml_client_config, AzureMLClientConfig)
+        if vc_config:
+            vc_config = validate_config(vc_config, VCConfig)
         self.azureml_client_config = azureml_client_config
+        self.vc_config = vc_config
```
Comment (trajepl) [inline olive/systems/azureml/aml_system.py:89]: simplify to?
`self.vc_config = validate_config(vc_config, VCConfig) if vc_config else None`

File: olive/systems/azureml/aml_system.py [hunk 4]
```diff
@@ -278,12 +293,20 @@ def _create_step(
         parameters.append("--pipeline_output ${{outputs.pipeline_output}}")
 
         cmd_line = f"python {script_name} {' '.join(parameters)}"
+        resources = None
+        if vc_config:
+            resources = {
+                "instance_type": vc_config.instance_type,
+                "instance_count": instance_count,
+                "properties": {"AISuperComputer": {**vc_config.extra_args} if vc_config.extra_args else {}},
```
Comment (trajepl) [inline olive/systems/azureml/aml_system.py:301]: Double check: is this `AISuperComputer` a fixed value always?

File: olive/systems/azureml/aml_system.py (hunk 3 above)
Comment (xiaoyu-work) [reply olive/systems/azureml/aml_system.py:89]: Done.

File: olive/systems/azureml/aml_system.py (hunk 4 above)
Comment (xiaoyu-work) [reply olive/systems/azureml/aml_system.py:301]: Yes I think so.

File: olive/systems/azureml/aml_system.py (hunk 3 above)
Comment (devang-ml) [reply olive/systems/azureml/aml_system.py:88]: There is no need to validate vc_config. Just pass it as it is resource. Let user provide proper dict.

Comment (xiaoyu-work) [reply olive/systems/azureml/aml_system.py:88]: @devang-ml this validation only turns a dict to VCConfig object. We only validate `instance_type` here.

**Comment by devang-ml** [comment]
> could the description be more details? I don't know the background so there is no way to review it.

+1

**Comment by devang-ml** [comment]
> > could the description be more details? I don't know the background so there is no way to review it.
> 
> +1

Something along the line "enable virtual clusters" ?

**Comment by xiaoyu-work** [comment]
> > > could the description be more details? I don't know the background so there is no way to review it.
> > 
> > 
> > +1
> 
> Something along the line "enable virtual clusters" ?

I synced with Mike offline about what this PR does. Also `virtual cluster` is not a public concept of Azure Machine Learning.

**Review by jambayk — APPROVED** [review]

**Pull request merged**